app/plugin.go

```diff
@@ -514,10 +514,15 @@ func (a *App) getRemotePlugins(filter *model.MarketplacePluginFilter) (map[strin
 	}
 
 	// Fetch all plugins from marketplace.
-	marketplacePlugins, err := marketplaceClient.GetPlugins(&model.MarketplacePluginFilter{
+	filter := &model.MarketplacePluginFilter{
 		PerPage:       -1,
 		ServerVersion: model.CurrentVersion,
-	})
+	}
+	if a.License() != nil {
+		filter.LicenseSkuShortName = a.License().SkuShortName
```

Comment:
I'm not sure this is the right approach -- it's too bad we can't just serialize the entire license and send that up.

Separately, and while I know the UX is being sorted out, I'd be very surprised if we /hide/ enterprise plugins altogether. I wonder if the "bit" of interest is whether or not the server binary was compiled with enterprise code or not (regardless of whether it's licensed)?

Reply:
> it's too bad we can't just serialize the entire license and send that up.

0/5 Include it into the body? I'm aware that a `GET` with a `body` is strange, but that might just fulfill the needs here.

@lieut-data The desired UX is, that EE-only plugins should only be shown to server that can actually run them. Otherwise admins might be frustrated by downloading a plugin that doesn't run on there server. What does an EE ready server indicate that might help here?

Reply:
> The desired UX is, that EE-only plugins should only be shown to server that can actually run them. Otherwise admins might be frustrated by downloading a plugin that doesn't run on there server. What does an EE ready server indicate that might help here?

The nuance here is that a server compiled with EE, but without a license, is /capable/ of running the plugins. We are actively pursuing UX that provides opportunities to prompt such an upgrade. By contrast, a server compiled without EE must be "physically replaced" even with a license in hand, and so can never run such a plugin. TLDR; I think showing enterprise-only plugins to an unlicensed, enterprise installation is desired at this time.

Reply:
@aaronrothschild  and I synced up offline:

The main goal of this change is to allow us to ship ee-only plugins via the Marketplace in the next three month, while not confusing admins who have don't have a license installed.

Due to the complexity in differentiating between E10 and E20, all EE-only plugin will work on both E10 and E20. There are no plans to add E20-only plugins for now.

While advertising enterprise-only plugins is an interesting idea, it's not feasible to implement for `v5.24.0`.

With all of that in mind the proposal is to have a `is_licensed` URL param. The Marketplace will omit any ee-only plugins if `is_licensed == false`.


@lieut-data Please let me know what you think about this.

Reply:
@hanzei 
> Due to the complexity in differentiating between E10 and E20, all EE-only plugin will work on both E10 and E20. There are no plans to add E20-only plugins for now.

@hanzei Small change to this based on an offline discussion yesterday: These will be E20 only plugins.  They should not work on E10 licences.  

Could we for now do a check on the "LDAP" feature for example - which is only in E20, then run the plugin based on that feature being licenced? 

Over tie we should add a feature called "E20 only plugins" to licences, and that is only available on E20 licences.

Reply:
1/5 - in the current (non-interactive) experience of installing plugins, exposing plugins that do not work seems like the wrong thing to do. I think it's fine if we currently only expose the enterprise plugins to _licensed_ customers.

In the future, we should make the installation of plugins more interactive, and at that point I agree it'd be benefitial to expose all available plugins; and if a user attempts to install one that requires a license - warn then. cc @aaronrothschild

Reply:
Based on the discussion above and offline I updated the code to:
- Send a `license_type` field that is `""`, `"E10` or `"E20"`. `"E20"` is send if the LDAP feature is enabled. That means if we change the way we map features to license types substantially, the information might be incorrect. But I don't see a better way to implement that check. @mkraft Thoughts?
- Send a `ee_ready` field that is `true` if the server was complied using the enterprise code. @lieut-data suggested adding this flag as it given us flexibility if we e.g. decide to show EE plugins to only `ee_ready` servers.


This approach given us maximum feasibility to decide later which plugins we want to show for which servers.